### PR TITLE
feat: Integrate Wan with multi-resolution DL

### PR DESCRIPTION
**Pull request by pthombre**
- Add text-to-video multiresolution dataloader: Introduces TextToVideoDataset and build_video_multiresolution_dataloader to support video training (Wan, HunyuanVideo) with bucket-based multiresolution sampling, replacing the legacy build_dataloader / MetaFilesDataset path.                                           
  - Refactor diffusion datasets with shared base class: Extracts common multiresolution logic (metadata loading, bucket grouping, dynamic batch sizing) into BaseMultiresolutionDataset, with TextToImageDataset and TextToVideoDataset as concrete implementations.                                                          
  - Refactor and rename dataloader APIs: Renames collate_fn_flux → collate_fn_text_to_image, build_flux_multiresolution_dataloader → build_text_to_image_multiresolution_dataloader, and extracts a shared _build_multiresolution_dataloader_core helper. Moves collate_fn_production and dataloader builders from sampler.py 
  to collate_fns.py.                                                                                                                                                                                                                                                                                                          
  - Add video collate function (collate_fn_video) with support for model-specific optional fields (e.g. text_mask, image_embeds).                                                                                                                                                                                             
  - Update example configs: Migrate Wan 2.1 and HunyuanVideo YAML configs to use the new video multiresolution dataloader; update Flux configs to use renamed image dataloader API.
  - Remove obsolete CI/CD nightly scripts and configs for Wan 2.1 pretrain.
  - Fix Flux attention backend configuration (use "flash" instead of "_flash_3_hub").
  - Fix and update unit tests for renamed APIs and new video dataloader.

### CODE REVIEW

File: nemo_automodel/_diffusers/auto_diffusion_pipeline.py [hunk 1]
```diff
@@ -89,7 +89,6 @@ class PipelineSpec:
             pipeline_cls: "FluxPipeline"  # Optional
             subfolder: "transformer"
             load_full_pipeline: false
-            enable_gradient_checkpointing: true
```
Comment (akoumpa) [inline nemo_automodel/_diffusers/auto_diffusion_pipeline.py:92]: was this not working?

Comment (pthombre) [reply nemo_automodel/_diffusers/auto_diffusion_pipeline.py:92]: The diffusers implementation of gradient checkpointing is not compatible with a model parallelized with fsdp
